### ISSUE TITLE
Buildings missing 3D models

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#1256]
For future reproduction and animation, 3D blender models are needed for the following tribal buildings:

Barbarian building		headquarters_interim
Barbarian building		destroyed building
Barbarian building		reed_yard
Barbarian building		smelting_works

The following models are currently being worked on:
Barbarian building		lime_kiln

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-01-14T08:51:32Z
 * owner: sirver
 * assignee: widelands-media-dev
 * the launchpad url was https://bugs.launchpad.net/bugs/677337

**Comment by bunnybot** [comment]
(by alexiade)
Are the Empire buildings really missing? I didn't do the donkey farm, but I should have the brewery(this really should have been in the old media svn...) and the shipwreck somewhere... Ill take look later tonight, after work. Pitty about the barbarian buildings, they were lost before I came along. I recovered what I could.

**Comment by bunnybot** [comment]
(by alexiade)

**Comment by bunnybot** [comment]
(by alexiade)
This really should be in the reporstory, its named brewer.blend tho, not brewery.

**Comment by bunnybot** [comment]
(by chuckw20)
Alexia, you are right about the Empire brewer building.  My miss.  :P   Thanks for the digging up the shipwreck model!

If you'd like to try your hand at remodeling some barbarian buildings, your efforts will be greatly appreciated!  ;)

Always good to hear from you!

**Comment by bunnybot** [comment]
(by timo-wingender)
I pushed the empire donkeyfarm and the atlantean horsefarm.

Unfortunately these are not exactly the version as in the game currently. I did some small changes before rendering the images but then quit without saving. So there a slight differences.

**Comment by bunnybot** [comment]
(by chuckw20)
@Timowi #5:
Thanks, for pushing those blend files and letting me know about the differences from the renders in the game.  Everything that avoids having to start from scratch is good.  :)

**Comment by bunnybot** [comment]
(by chuckw20)
For reference, I have attached a graphic file that shows the current game images of the models needed.  

It also suggests existing models that could be used as templates or cannibalized for objects and materials.

**Comment by bunnybot** [comment]
(by alexiade)
The early barbarian models what we have got have the textures in a separate file that are linked into the models files. They all may as well need an overhaul because of the broken linking and use of a different render stage to be fully compatible.

**Comment by bunnybot** [comment]
(by chuckw20)
@Alexia #8:
Yes.  We still have the textures file plus another with various building parts.  It would likely be good to pursue check, and if needed, re-render the existing models (older ones anyway) to ensure the use of a rendering stage consistent with the rest of the game.  I've always thought the barbarian headquarters and know for certain some of the smaller buildings had been rendered at a slightly different camera angle than what we are now using.

**Comment by bunnybot** [comment]
(by chuckw20)
Here is my re-model of the barbarian headquarters.  There will be animation and the penants will be in playercolor.

Comments?

**Comment by bunnybot** [comment]
(by alexiade)
looks really nice:)

**Comment by bunnybot** [comment]
(by chuckw20)
Final version of the headquarters building pushed to trunk in rev#5750.

**Comment by bunnybot** [comment]
(by chuckw20)
Final version of the sentry building pushed to trunk in rev#5751.

**Comment by bunnybot** [comment]
(by chuckw20)

**Comment by bunnybot** [comment]
(by chuckw20)
After discussing it in the Graphics Forum, it was agreed to show construction of a building in 4 frames to give a visual indicator to approximate its completion.  Subsequently I have modified the sentry building to have 4 frames to depict its build progress.  The changes were pushed to trunk in rev#5752.

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian stronghold pushed to trunk in rev #5759.

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian barrier pushed to trunk in rev #5763.

**Comment by bunnybot** [comment]
(by chuckw20)

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian donjon pushed to trunk in rev #5765.

**Comment by bunnybot** [comment]
(by chuckw20)

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian battlearena pushed to trunk in rev #5773.

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
i guess i just did something i don´t understand, anyway i think it is good ^^

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
got the oremine and the goldmine online i guess

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
oremine allmost looks the same... now i hope i will someday understand how this worked xD

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
deep coalmine for christmas okay? got ore and goldmines on the way...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
argh somehow moving the .blends was no good at all, lets see what i can do about the materials,maybe my mines will all update because of lost materials xD

**Comment by bunnybot** [comment]
(by chuckw20)
@kingcreole:
First, congratulations on your first successful pushes to Launchpad!  That is a big hurdle to overcome.

Some observations on the mines:
1) The path on the ground plane that leads to the mine entrance does not line up with the approach line used by the workers.  Please adjust this and reposition any objects (like piles of ore) that may obstruct the path.

2) Add a fourth frame to the build sequence animation of all buildings.  It should represent the completed building.  Currently without it the building gets its roof instantaneously upon completion of the 3rd frame.  This fourth frame can be a copy of the first idle frame, but it should not be omitted.

3) The transition to the enhanced deep mine also needs attention to its first and last build frames.  When I clicked the enhance button, the roof disappeared long before the builder got to the site.

3) Double check the hotspots for working v.s. idle animations of the small mines.  To my eyes the iron ore and gold mines appear to change position.

4) The lighting isn't behaving naturally.  The flickering is good, but it appears to emit from a point to the side of the shaft and it bleeds onto the roof beyond the canopy.  I would like to see the effect either emanating from the mine shaft itself or from the doorway of the building.

All in all a great first effort, but not quite ready for merging.

**Comment by bunnybot** [comment]
(by chuckw20)
Looks like I should double-check my item numbering before clicking on "POST COMMENT"...   :)

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
^^ i´ll rerender anyway, had problems with the materials linked.
1: at least you didn´t notice, that the path changes when the build animation is finished xD, got it on my list and now i can finaly edit all of those groundplanes at once
2:jup, gotta copypasta that.
3.1: copypasta as well (maybe i´ll write a little script and run it 3 times for all of those mines xD)
3.2: OO oops, just checked it: oremine and goldmine got that problem, coalmine is fine... i tried those mines too and didn´t notice anything xD
4: its a lamp with raytracing disabled ^^` had a groundplane inside that mine and the light beneath that groundplane, maybe i´ll take it above the groundplane and enable raytracing...
5: got imagemagick working, that thing might speed it all up...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
got it all re-rendered now only copyin, pastin, renamin... at the deep mines i erased one building step, the b_00 is the idle of the previous mine, the b_03 is the idle of the deep mine wich only leaves plave for 2 building steps... (was boring animationframe anyway)

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
got it all ready pushin it...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
pushed normal and deep mines sooooon i will start working on some deeper mines

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
are those mines allright now? since i pushed them i didn´t get no feedback at all...

**Comment by bunnybot** [comment]
(by sirver)
king, i have not heard from chuck which is usually a good sign that he is busy with his RL. I myself am quite busy atm, so I couldn't check your stuff. Please be a little patient, we will not overlook this contribution and all others you might make. :)

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
jeehaw that close milestone is awesome ^^ gotta get working, got the deeper oremine allmost done (playercolor and conf file missing) soon i´ll test if it works out cuz of those surrounding roads and get the others working too...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
-.- just noticed, that mine is too big to be a deeper mine... i guess i´ll redesign that whole deeper mine style...

**Comment by bunnybot** [comment]
(by nasenbaer)
@kingcreole: just wanted to let you know, that I really like your mines :)

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
^^ thank you, gotta get those mines done soon, i guess i´ll try some helping grid to get that mine sized correctly...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
deep mines done and pushed, working on some bugreports...

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
fixed bugs, i guess those mines are ready for takeoff :D

**Comment by bunnybot** [comment]
(by hjd)
kingcreole: Based on the screenshots here and on the forum, I think your mines look great! I look forward to seeing them in the game. :)

**Comment by bunnybot** [comment]
(by chuckw20)
kingcreole's barbarian gold, iron and coal mines have been merged with trunk in rev 5824!

Great work!!!

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
added granitemine :) no bugs SHOULD occur, tested on my machine, roads should not intersect mines either... i guess its okay, coments? :)

**Comment by bunnybot** [comment]
(by chuckw20)
kingcreole, please review the b_00 png file of the granitemine.  The scale and placement of the timbers differs drastically from the reset of the frames.  That must be corrected before we can put it in the game.

Otherwise, the rest looks good to go.  Nice work!

**Comment by bunnybot** [comment]
(by chuckw20)
Make that "differs drastically from the rest of the frames."

**Comment by bunnybot** [comment]
(by chuckw20)
Updated the buildings To Do Graphic

Only 17 more to go!  :)

**Comment by bunnybot** [comment]
(by chuckw20)
Make that 18.  :)

**Comment by bunnybot** [comment]
(by chuckw20)
Scratch that. It's 17.  I forgot to cross off the donjon.

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
jop, when i tested today i saw that problem with the huge pieces of wood at b_00 XD gonna change and push tomorrow

**Comment by bunnybot** [comment]
(by chuckw20)
Added kingcreole's new granitemine in trunk rev#5830.  Great job!

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
:) next thing on my list : well

**Comment by bunnybot** [comment]
(by hjd)
I took the liberty of updating the description, marking the well as a work in progress. I thought it might be a good idea to separate the list of models needed and those currently worked on to avoid duplicate work.

**Comment by bunnybot** [comment]
(by chuckw20)
@ Hans #52   
Thanks for updating the description.  Listing the works in progress is a very good idea to prevent duplication of effort.  (I added my current models to the list.)

@kingcreole   
Nice job on the well. It's work animation tests well (no pun intended), so I'll merge your current branch into the trunk.   
At your option, the only changes I'd like to see in the renders are:

    1) to the placement of the pennant (maybe to where you have put it in the blender model) so that it doesn't obscure and
    2) give the pennant some animation, but it's fine as it stands.

Regarding the blender file, I would prefer that the model did not differ from the rendered images. Please correct one or the other and I'll merge the modification(s) when ready.

**Comment by bunnybot** [comment]
(by chuckw20)
kingcreole's barbarian well has been added to the main trunk in rev#5824.

**Comment by bunnybot** [comment]
(by chuckw20)
Correction:  kingcreole's barbarian well has been added to the main trunk in rev#5847.

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
okay :D should i still change the well even though its in the trunk?

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
about the .blend file, how does it differ from the rendered images?

**Comment by bunnybot** [comment]
(by chuckw20)
Compare this attached render from the blend file with your frames.

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
do the mines look okay or will i have to re-push them?

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
back btw ;)

**Comment by bunnybot** [comment]
(by chuckw20)
@kingcreole#59  The mines are in the game as of rev #5830. (see post #50)  No need to push them again. :)

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
pushed that well again with the images packed, i guess that problem is solved...
anyway easter hollidays now so letze go! get those babarian buildings remodeled :D

**Comment by bunnybot** [comment]
(by gerrit-familiegrosskopf)
that "letze go!" only created an epic fail :D anyway if i find the time i´ll work on them (thats soo pessimistic, it can´t fail)

**Comment by bunnybot** [comment]
(by chuckw20)
The fortress and citadel models are complete and in with rev. #5975

**Comment by bunnybot** [comment]
(by chuckw20)

**Comment by bunnybot** [comment]
(by chuckw20)
The new trainingscamp model was introduced in rev #5985

**Comment by bunnybot** [comment]
(by chuckw20)
The battlearena model has been added in rev #5989

**Comment by bunnybot** [comment]
(by chuckw20)

**Comment by bunnybot** [comment]
(by chuckw20)
The new metalworks, axefactory and warmill  have been added in rev #6034.

**Comment by bunnybot** [comment]
(by sirver)
Setting to incomplete for bug sweeping.

**Comment by bunnybot** [comment]
(by sirver)
Setting to incomplete for bug sweeping.

**Comment by bunnybot** [comment]
(by chuckw20)
Work has started again on models for the barbarian buildings.  Attached is a sample of the warehouse.  I've kept the idea of a stabbur like the original.

I'll post a screen shot below

**Comment by bunnybot** [comment]
(by chuckw20)
Here is a screenshot of the building in the game with player color.

Comments?

**Comment by bunnybot** [comment]
(by chuckw20)
Here is a screenshot of the building in the game with player color.

Comments?

**Comment by bunnybot** [comment]
(by chuckw20)
Sample #2 - Here is a tweaked version of the warehouse with a little less player color. :)

**Comment by bunnybot** [comment]
(by chuckw20)
Sample #2 shown in game

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian warehouse implemented in trunk bzr rev7361.

**Comment by bunnybot** [comment]
(by gunchleoc)
I like it :)

An idea for the cross-beams in the front - how about making them a bit thicker depth-wise (front-to-back)? It would also look cool to have some sculpted ends, like horses' heads, for example

https://de.wikipedia.org/wiki/Pferdek%C3%B6pfe_%28Giebelschmuck%29

**Comment by bunnybot** [comment]
(by chuckw20)
I have amended the tavern/inn/big_inn models to hopefully make them easier to recognize.

These shots show them in their idle state.  All three have animation for their work and build cycles.

I have tried to incorporate suggestions made in the forum https://wl.widelands.org/forum/topic/1633 (i.e. lighting the windows and doorways, emphasizing more identifiable features.)

**Comment by bunnybot** [comment]
(by chuckw20)
Branch https://code.launchpad.net/~chuckw20/widelands/barb_tavern_inns has been pushed for testing and merge.

**Comment by bunnybot** [comment]
(by chuckw20)
This is an image of the new tavern, inn and big_inn in the game in their working cycles.

**Comment by bunnybot** [comment]
(by chuckw20)
The new tavern, inn and big_inn have been pushed to trunk in r7377. 
http://bazaar.launchpad.net/~widelands-dev/widelands/trunk/revision/7377

**Comment by bunnybot** [comment]
(by chuckw20)
Attached is a graphic of the charcoal_kiln that I have modeled.

**Comment by bunnybot** [comment]
(by gunchleoc)
Nice :)

**Comment by bunnybot** [comment]
(by chuckw20)
Model #6 rescales and moves the building and burning mound.

**Comment by bunnybot** [comment]
(by chuckw20)
The new charcoal_kiln has been pushed to trunk in r-7382.  I tweaked some things and let her rip. :)

**Comment by bunnybot** [comment]
(by gunchleoc)
Can you please give files the same names as their animations in the future? e.g.

big_inn_b_00.png => building_00.png
big_inn_i_00.png => idle_00.png
big_inn_w_00.png => working_00.png

We already know which building it is from its directory, so I'd like some more pertinent information in the filename.

Don't worry about the ones you have already done, I will fix them up in the one_tribe branch anyway.

**Comment by bunnybot** [comment]
(by chuckw20)
@#87 - Happy to oblige.  I apologize for any ignorance (or more likely forgetfulness) on my part regarding the file naming conventions.  I appreciate the notification.

**Comment by bunnybot** [comment]
(by gunchleoc)
No problem. The files used to be named just like you did; this is a very recent change that you couldn't have known about.

**Comment by bunnybot** [comment]
(by chuckw20)
@#89 - Gee, I drop out for a couple of years and everything is changed.  I'm glad the development continues. :)

**Comment by bunnybot** [comment]
(by gunchleoc)
The menu.png for the Charcoal Kiln is too big. Check the Building statistic to see what I mean.

**Comment by bunnybot** [comment]
(by chuckw20)
@91 - Thanks for bringing that to my attention.  Will fix ASAP.

**Comment by bunnybot** [comment]
(by chuckw20)
@91 - fix pushed in r7419.

**Comment by bunnybot** [comment]
(by chuckw20)
New barbarian wood hardener's hut committed in r7422.

**Comment by bunnybot** [comment]
(by chuckw20)
The media branch does contain the .blend file for the Atlantean shipyard.

**Comment by bunnybot** [comment]
(by gunchleoc)
We were just changing some code and noticed that there is only an idle animation

https://code.launchpad.net/~widelands-dev/widelands/bug-1442945/+merge/258712

If this isn't the correct bug, is there a todo-list for stuff like that?

**Comment by bunnybot** [comment]
(by chuckw20)
This bug is intended merely to track the creation of 3D models for those buildings that currently have none.  

I think missing/desired animation is a matter that, for now should be addressed on a per building/incident basis.  So please submit a separate bug report about the atlantean shipyard animation.  Thx.

**Comment by bunnybot** [comment]
(by gunchleoc)
Done https://bugs.launchpad.net/widelands-media/+bug/1454168

**Comment by Alexia** [comment]
You just spammed me with tons of email due to not sanitizing issue descriptions and comments during your import.

**Comment by gunchleoc** [comment]
Note: Many of the barbarian buildings have no floor texture (in case it has no closed door). Thus the texture of the surrounding floor (Like grass, mountain etc.) is also inside the building.
No matter if this was designed purposely or not, some textures should be edited.
